### DISCUSSION TITLE
ArcGIS Server 11.3 Bug - Return Count Only

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#212] [Q&A]
Hallo Jürgen!

Der neue ArcGIS Server 11.3 (und auch 11.4) liefert ja bei Return Count Only und projection on the fly (DB = 31259, Map = 3857) falsche Ergebnisse. Die Transformation wird im AGS nicht berücksichtigt. 
Da es unklar ist, wann von Esri ein Patch dafür bereitgestellt wird, wurde nach einem Workaround gesucht. 
Den gibt es auch: Mit **datumtransformation=1618** als zusätzlicher Parameter bekommt man das korrekte Ergebnis.

Ist es möglich im WebGIS beim Request den zusätzlichen Parameter **datumtransformation=** mitzugeben, damit beim Identify mit AGS 11.3 korrekte Ergebnisse geliefert werden?

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Hallo Andi,

Das gibt es schon:
Du kannst beim Dienst im CMS auf **Erweiterte Einstellungen** gehen:

![image](https://github.com/user-attachments/assets/29283768-e439-4ee9-8b32-d5a7adbb3245)

Dort gibt es dann ein Punkt **Kartemprojektion**:

![image](https://github.com/user-attachments/assets/a936385c-3e9e-4dd5-a989-c54d1558f36e)

Wenn du hier einen Wert eingibst, sollte der genau so mit übergeben werden. Probier das bitte einmal, falls es nicht hinhaut, müsste ich noch einmal mit einen Dienst von euch testen.

LG

**Comment by Andreas-Lampert** [comment]
Hallo Jürgen!

Danke für die Info! Ich habe gerade versucht den Wert 1618 einzutragen und ich bekomme eine Fehlermeldung.
![image](https://github.com/user-attachments/assets/2bb046ae-fb37-48c4-9885-1bc69ac5605e)

Im *.xml sehe ich den Parameter nicht.
![image](https://github.com/user-attachments/assets/0c6c3172-049a-4c04-994d-efc9d7bebed0)

Kann ich den Wert für die Tests irgendwie direkt ins *.xml eintragen? Wie müsste der Parameter genau eingetragen werden?
Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Ok, man kann hier anscheinend mehrere Werte eingeben. Also muss man ein Array eingeben :/

Probier es mit ``[1618]``

**Comment by Andreas-Lampert** [comment]
Danke, mit dem Array hat es funktioniert. Es hat im WebGIS allerdings leider keine Änderung bewirkt. Beim Identify mit Count Only kommt kein Ergebnis bzw. das falsche Ergebnis. 
Die Server sind derzeit nur intern verfügbar, daher kann ich dir kein Service mit AGS 11.3 zur Verfügung stellen. Wir können aber sehr gerne gemeinsam drauf schauen (falls dir das irgendwas bringt). Sonst muss ich schauen, dass der Server irgendwie zumindest temporär von Extern aufrufbar ist. 

Lg, Andi

**Comment by gis-eni** [comment]
Ok, ein Problem ist mir noch aufgefallen. Früher konnten man anscheinend mehrere Transformationen übergeben und der Paramater hieß ``datumTransformations``. Esri scheint sich hier auch in der Doku nicht ganz einig zu sein ;)
https://developers.arcgis.com/rest/services-reference/enterprise/query-map-service-layer/
Wenn man eine Query über die AGS REST Oberfläche abschickt, verwenden sie ``datumTransformation`` was logisch und richtig erscheint. 

Ich hab das jetzt einmal auch im WebGIS Code durchgecheckt und da gabs noch ``datumTransformations``. Sollte jetzt dann ausgebessert sein und wird ab jetzt bei jedem Request mitgeschickt (falls parametriert):

![image](https://github.com/user-attachments/assets/a6bb499c-6909-40d9-9d01-da9a630c00eb)

Aus Gründen der Abwärtskompatibilität muss man in CMS aber nimmer noch ein Array angeben, wobei dann nur der erste Wert berücksichtigt wird:

![image](https://github.com/user-attachments/assets/d8233649-e8a8-4061-a4a4-2ca2ceb54218)

Für dich sollte genau diese Einstellung reichen (keine ProjektionMethod, dass sollte aus dem Dienst ersichtlich sein).

Ich melde mich noch einmal bei dir, sobald ein Build zum Testen für dich bereit ist.

**Comment by Andreas-Lampert** [comment]
Hallo Jürgen!

Vielen Dank! Es funktioniert mit der Version 6.24.4501.
